### ISSUE TITLE
emulator throwing error MODULE_NOT_FOUND with --inspect-functions

### ISSUE THREAD

**Issue by abuharsky** [firebase/firebase-tools#2380]
### [REQUIRED] Environment info

**firebase-tools:** 8.4.3

**Platform:** mac os catalina 10.15.5
node -v 
v12.18.1

visual studio code Version: 1.46.1
### [REQUIRED] Test case

1) created new project with 1 firebase function "hello_word"
2) when I have run "firebase emulators:start --only firestore,functions"
al works perfect

3) when I run "firebase emulators:start --only firestore,functions --inspect-functions"
I've got error:
---

functions: Watching "/Users/username/Documents/Firebase/myproject/purchases/functions" for Cloud Functions...
>  internal/modules/cjs/loader.js:983
>    throw err;
>    ^
>  
>  Error: Cannot find module '/Users/username/Documents/Firebase/myproject/purchases/functions/--inspect=localhost:9229'
>      at Function.Module._resolveFilename (internal/modules/cjs/loader.js:980:15)
>      at Function.Module._resolveFilename (pkg/prelude/bootstrap.js:1346:46)
>      at Function.Module._load (internal/modules/cjs/loader.js:862:27)
>      at Function.Module.runMain (pkg/prelude/bootstrap.js:1375:12)
>      at internal/main/run_main_module.js:18:47 {
>    code: 'MODULE_NOT_FOUND',
>    requireStack: []
>  }
---

### [REQUIRED] Steps to reproduce

run "firebase emulators:start --only firestore,functions --inspect-functions"

[firebase-debug-without-inspect.log](https://github.com/firebase/firebase-tools/files/4806781/firebase-debug-without-inspect.log)

[firebase-debug-with-inspect.log](https://github.com/firebase/firebase-tools/files/4806782/firebase-debug-with-inspect.log)

**Comment by samtstern** [comment]
This looks like a duplicate of this issue:
https://github.com/firebase/firebase-tools/issues/2320

Did you install the Firebase CLI through `npm` or through the `curl` script?

**Comment by abuharsky** [comment]
> This looks like a duplicate of this issue:
> #2320
> 
> Did you install the Firebase CLI through `npm` or through the `curl` script?

I have installed 7.7.0 with node, but later updated with curl to 8.4.3

**Comment by samtstern** [comment]
@abeisgoat any thoughts on this one?  Should it be fixed in the latest Firepit binaries?

**Comment by quantuminformation** [comment]
Just found this myself


![image](https://user-images.githubusercontent.com/216566/85923456-77514080-b88b-11ea-86b7-0707667877e2.png)

**Comment by quantuminformation** [comment]
I used the curl script

**Comment by quantuminformation** [comment]
Just quit the emulator and checked the version again and saw this again:
```
firebase -V
⚠  emulators: emulators failed to shut down cleanly, see firebase-debug.log for details.
8.4.3
```

also seeing this one again

```
i  functions: Beginning execution of "listUsers"
>  Error listing users: FirebaseAuthError: //cloud.google.com/docs/authentication/. Raw server response: "{"error":{"code":403,"message":"Your application has authenticated using end user credentials from the Google Cloud SDK or Google Cloud Shell which are not supported by the identitytoolkit.googleapis.com. We recommend configuring the 

```
but I have creds here I made yesterday
![image](https://user-images.githubusercontent.com/216566/85923878-f300bc80-b88e-11ea-9418-e81b650df3ce.png)


new issue(s)?

**Comment by samtstern** [comment]
@quantuminformation your error message is not related to this issue:

```
>  Error listing users: FirebaseAuthError: //cloud.google.com/docs/authentication/. Raw server response: "{"error":{"code":403,"message":"Your application has authenticated using end user credentials from the Google Cloud SDK or Google Cloud Shell which are not supported by the identitytoolkit.googleapis.com. We recommend configuring the 
```

That is https://github.com/firebase/firebase-tools/issues/1371

**Comment by AndriiTsok** [comment]
We are experiencing the same issue running emulator with: --inspect-functions attribute.

**Comment by mindful-labs** [comment]
Any update on this issue? My team is experiencing the same problem and it makes debugging local firebase functions impossible. Please prioritize this.

**Comment by quantuminformation** [comment]
Pls try updating your tools

**Comment by manaschauhan** [comment]
I'm also experiencing the same problem and makes it impossible to debug local firebase functions.

**Comment by shrestaz** [comment]
For me, I had this issue when Firebase CLI was installed with cURL.
I uninstalled Firebase CLI and installed it again using `npm install -g firebase-tools` and it works without issue.

**Comment by blasco** [comment]
Experiencing the same issue when using ` --inspect-functions`

**Comment by blasco** [comment]
Uninstalling and installing (now I have version 8.12.1) seems to solve the issue

**Comment by joaoeudes7** [comment]
@blasco yeah, after reinstall works perfect! kkk thanks

**Comment by samtstern** [comment]
@abeisgoat investigated this, I'll summarize what he told me.

It turns out this is only an issue with the standalone Firebase Tools binary (installed via `curl`) and not an issue with the `npm` install.  So if this is essential to you, install `firebase-tools` via `npm install -g firebase-tools` and it should work.

The reason this does not work is because we use [`pkg`](https://github.com/vercel/pkg) to embed `node` inside the standalone binary.  For security reasons this embedded `node` install blocks all Node debugging via the `--inspect` flag.  We may be able to find a way around this, but it will be a pretty big change so it's not likely to happen very soon.

**Comment by joaoeudes7** [comment]
@samtstern the installation command via curl could be a script that makes a global installation call via npm or yarn, and so it would partially solve this problem, after all, most developers for firebase cloud functions use typescript or javascript, so it is common that has some package managers of the type

**Comment by samtstern** [comment]
@joaoeudes7 you'd be surprised!  We had to make the `curl` version specifically because so many of our developers had issue with Node or NPM on their machine or had never used those tools before Firebase.   So that's why we have two separate options: lightweight install with `npm` or standalone install with `curl`.

**Comment by joaoeudes7** [comment]
@samtstern Really amazing! I ended up installing the curl version later because of the documentation in the beta test forum because I figured that the version would not be available on NPM.
Perhaps a possible solution would be to get out of `pkg`, this problem has existed since 2017, and who knows [nbin](https://github.com/cdr/nbin) can handle all cases well

**Comment by abeisgoat** [comment]
So good news! This is actually not as big of change as we thought to fix. Past @abeisgoat wrote a bunch of the code needed to fix this and didn't already know. As @samtstern said, in the single-binary releases the version of node packaged has some security restrictions which break function inspection, however the emulator is actually smart enough to discover other node installs for you, you just have to install them!

I've just submitted a small patch #2740 which will tell the emulator to favor any version of node installed locally via npm (`npm i node@10 --save-dev`) so it'll use that over the version of node packaged with the single binary. The next step is to catch this error before it happens and encourage devs to pin a version of node (via the above command) and let the emulator discover it over the built-in node runtime.

**Comment by abeisgoat** [comment]
This is fixed in #2740, the CLI will now tell you how to avoid this problem
